### ISSUE TITLE
Do not support UTF-8

### ISSUE THREAD

**Issue by SLyHuy** [wwwtyro/cryptico#12]
plaintext: "Mưa nắng là chuyện của mày, tao đéo quan tâm!!! Được không hả?"

after encrypt & decrypt: "MLa nZng là chuyn ca mày, tao ¢éo quan tâm!!! Sc không hÈ?"

**Comment by jpfox** [comment]
I've just done a correction on my repository about utf-8 : https://github.com/jpfox/cryptico

**Comment by SLyHuy** [comment]
Thank you.
What's your solution? I'm using: Convert UTF-8 to ASCII, Encrypt-Decrypt - Convert ASCII to UTF-8. by using this script: http://stackoverflow.com/a/834529/2480623

**Comment by jpfox** [comment]
I use `unescape(encodeURIComponent(string))` and `decodeURIComponent(escape(uriencoded))` to convert text before AES encryption and after decryption. It permits to switch easily to single byte encoding.

I made just few changes on original code : 
https://github.com/jpfox/cryptico/commit/73db107f8e1e0d59696e1aece9e7a2bc82268707

**Comment by b6g9** [comment]
Thank jpfox !

**Comment by Elmapul** [comment]
@jpfox  this will work with binary data instead of text?

**Comment by jpfox** [comment]
@Elmapul no, I don't think so. If you need a project using password-derived-key to encrypt files, have a look to [miniLock](https://minilock.io/)

**Comment by beenotung** [comment]
@jpfox would you create a pull request to the upstream someday? otherwise I may do the chord.

**Comment by jpfox** [comment]
You can use commit in previous comment to create a PR if you want. No problem for me.
